### PR TITLE
[Samples][42.scaleout] Add main structure + README + Store interface

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: samples/42.scaleout/src/main/resources/log4j2.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (fran893) [inline samples/42.scaleout/src/main/resources/log4j2.json:1]: Is this file correct? what is its usage?

**Review by FedericoBernal — APPROVED** [review]
LGTM

File: samples/42.scaleout/pom.xml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        <scope>compile</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.microsoft.bot</groupId>
+        <artifactId>bot-dialogs</artifactId>
+        <version>4.6.0-preview9</version>
+        <scope>compile</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.microsoft.bot</groupId>
```
Comment (fran893) [inline samples/42.scaleout/pom.xml:94]: Is this the correct dependency? I checked the [csproj](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/42.scaleout/ScaleoutBot.csproj#L10) in C# and it is using the azure storage blob dependency and not the one from bot builder.

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/42.scaleout/pom.xml (hunk 2 above)
Comment (Batta32) [reply samples/42.scaleout/pom.xml:94]: You are totally right, thanks for pointing this

File: samples/42.scaleout/src/main/resources/log4j2.json (hunk 1 above)
Comment (Batta32) [reply samples/42.scaleout/src/main/resources/log4j2.json:1]: You are totally right, thanks for pointing this

File: samples/42.scaleout/pom.xml (hunk 2 above)
Comment (Batta32) [reply samples/42.scaleout/pom.xml:94]: You are totally right, thanks for pointing this

File: samples/42.scaleout/src/main/resources/log4j2.json (hunk 1 above)
Comment (Batta32) [reply samples/42.scaleout/src/main/resources/log4j2.json:1]: The file is correct, I'm not quite sure what's the goal right now, this is part of all the samples

**Review by fran893 — APPROVED** [review]

**Pull request merged**